dbt_coding_conventions.md

```diff
@@ -47,33 +52,38 @@
 ``` sql
 with events as (
 
-	...
+    ...
 
 ),
 
 -- CTE comments go here
 filtered_events as (
 
-	...
+    ...
 
 )
 
 select * from filtered_events
 ```
 
-## SQL Style Guide
+## SQL style guide
 
 - Indents should be four spaces (except for predicates, which should line up with the `where` keyword)
 - Lines of SQL should be no longer than 80 characters
 - Field names and function names should all be lowercase
 - The `as` keyword should be used when projecting a field or table name
 - Fields should be stated before aggregates / window functions
-- Ordering and grouping by a number (eg. group by 1, 2) is ok
+- Ordering and grouping by a number (eg. group by 1, 2) is preferred. Note that if you are grouping by more than a few columns, it may be worth revisiting your model design.
 - When possible, take advantage of `using` in joins
 - Prefer `union all` to `union` [*](http://docs.aws.amazon.com/redshift/latest/dg/c_example_unionall_query.html)
+- Avoid table aliases (especially initialisms).
+- If joining two tables, _always_ prefix your column names with the table alias.
+- Be explicit about your join (i.e. write `inner join` instead of `join`). `left joins` are normally the most useful, `right joins` usually mean that you've chosen to select `from` the wrong table (and you should instead change which table you select `from` and which one you `join` to).
+
 - *DO NOT OPTIMIZE FOR A SMALLER NUMBER OF LINES OF CODE. NEWLINES ARE CHEAP, BRAIN TIME IS EXPENSIVE*
 
 ### Example SQL
+Note that while this demonstrates good SQL style, it does not represent good modeling, since the `final` CTE both aggregates and joins!

```

Comment:
is that a problem? I frequently don't do this but at the same time i've never heard anyone say that this is _bad_. why do you believe it is bad?

Reply:
I have an aversion to grouping by any columns other than those that represent the grain of the model

Reply:
Hmm. I get where you're coming from but we don't list that as an explicit standard here and, while I support that opinion generally, it feels overly prescriptive for us to have that as a standard. I'm therefore hesitant to say that this is "not good modeling".